File: LiveDataHandler/models/auditlog.ts
/*
 * Audit log data model.
 *
 * @author Michel Megens
 * @email  michel@michelmegens.net
 */

export interface AuditLog {
    route: string;
    method: number;
    ipAddress: string;
    authorId: string;
    timestamp: Date;
}

export enum RequestMethod {
    HttpGet,
    HttpPost,
    HttpPatch,
    HttpPut,
    HttpDelete,

    WebSocket,

    MqttTcp,
    MqttWebSocket,
    Any
}

File: SensateIoT.Platform/SensateIoT.Platform.Network/SensateIoT.Platform.Network.Data/Models/AuditLog.cs
/*
 * Audit logging model.
 *
 * @author Michel Megens
 * @email  michel@michelmegens.ne
 */

using System;
using System.Net;

using SensateIoT.Platform.Network.Data.Enums;

namespace SensateIoT.Platform.Network.Data.Models
{
	public class AuditLog
	{
		public long Id { get; set; }
		public string Route { get; set; }
		public RequestMethod Method { get; set; }
		public IPAddress Address { get; set; }
		public string AuthorId { get; set; }
		public DateTime Timestamp { get; set; }
	}
}
